File: Mixter/Domain/Identity/SessionHandler.cs
using Mixter.Domain.Identity.Events;
using Mixter.Infrastructure;

namespace Mixter.Domain.Identity
{
    public class SessionHandler : IEventHandler<UserConnected>
    {
        private readonly ISessionsRepository _repository;

        public SessionHandler(ISessionsRepository repository)
        {
            _repository = repository;
        }

        public void Handle(UserConnected evt)
        {
            _repository.Save(new SessionProjection(evt));
        }
    }
}

File: app/Domain/Identity/SessionHandler.php
<?php

namespace App\Domain\Identity;

class SessionHandler {
    /**
     * @var ISessionProjectionRepository
     */
    private $sessionRepository;

    public function __construct(ISessionProjectionRepository $sessionRepository) {
        $this->sessionRepository = $sessionRepository;
    }

    public function handleUserConnected(UserConnected $userConnected)
    {
        $this->sessionRepository->save(
            new SessionProjection($userConnected->getUserId(), $userConnected->getSessionId()));
    }

    public function handleUserDisconnected(UserDisconnected $userDisconnected)
    {
        $this->sessionRepository->remove($userDisconnected->getSessionId());
    }
}